File: semantic-conventions/src/main/java/events/boudicca/EventCategory.java
package events.boudicca;

import java.util.Set;

public enum EventCategory {

    MUSIC(Set.of("konzert", "concert", "alternative", "singer/songwriter", "party", "songwriter/alternative", "musik")),
    ART(Set.of("kabarett", "theater", "wissenskabarett", "provinzkrimi", "comedy", "figurentheater", "film", "visual comedy", "tanz", "performance", "musiklesung", "literatur", "museen", "museum", "ausstellung", "musical")),
    TECH(Set.of("techmeetup"));

    public final Set<String> types;

    EventCategory(Set<String> types) {
        this.types = types;
    }


}

File: boudicca.base/semantic-conventions/src/main/kotlin/base/boudicca/model/EventCategory.kt
package base.boudicca.model

import java.util.*

enum class EventCategory(val types: Set<String>) {
    //ALL and OTHER are special cases needing special care
    ALL(emptySet()),
    OTHER(emptySet()),
    MUSIC(setOf("konzert", "concert", "alternative", "singer/songwriter", "party", "songwriter/alternative", "musik")),
    ART(
        setOf(
            "kabarett",
            "theater",
            "wissenskabarett",
            "provinzkrimi",
            "comedy",
            "figurentheater",
            "film",
            "visual comedy",
            "tanz",
            "performance",
            "musiklesung",
            "literatur",
            "museen",
            "museum",
            "ausstellung",
            "musical"
        )
    ),
    TECH(setOf("techmeetup", "technology", "technologie"));

    companion object {
        fun getForType(type: String?): EventCategory? {
            if (type == null) {
                return null
            }
            val lowerType = type.lowercase(Locale.getDefault())
            for (category in entries) {
                for (categoryType in category.types) {
                    if (lowerType.contains(categoryType)) {
                        return category
                    }
                }
            }
            return null
        }
    }
}